proto/marginal_state.proto

```diff
@@ -10,8 +10,9 @@ message MarginalState {
     oneof val {
       UnivLSState univ_ls_state = 1;
       MultiLSState multi_ls_state = 2;
+      LinRegUnivLSState lin_reg_univ_ls_state = 3;
     }
-    int32 cardinality = 3;
+    int32 cardinality = 4;
```

Comment:
Changing the numbering of proto messages is kind of bad practice, since already serialized chains cannot be deserialized with the new message structure.

If instead you keep the old numbering and add fields with increasing numbers, everything works fine.